step/src/main/java/com/google/sps/MessageHandler.java

```diff
@@ -65,9 +65,10 @@ public static void addMessage(Message m) {
             firstUserRecipientFilter, secondUserSenderFilter
     ))));
 
-    Query query = new Query("Message").setFilter(compositeFilter).addSort("timestamp", SortDirection.ASCENDING);
+    Query query = new Query("Message").setFilter(compositeFilter);
+    //query.addSort("timestamp", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
-    List<Message> messages = new ArrayList<>();
+    ArrayList<Message> messages = new ArrayList<>();
```

Comment:
Why did you choose to reduce the scope of the messages object? In general, it's better to have a higher scope so that the return object is more accessible.

Reply:
I could not think of a reason why but when I changed it back it had errors in some test cases. I might try to look over to see when it started having the error